.github/workflows/build.yml

```diff
@@ -23,32 +23,10 @@ jobs:
           ./autogen.sh
           ./configure CXXFLAGS="-Wall -fdiagnostics-color=always"
           make -j "$(nproc)" |& tee build.log
-      - name: Summarize warnings
+      - name: "Summarize warnings"
         run: python3 ./scripts/count-warnings.py build.log
 
-  build_gcc8:
-    name: "GCC 8 (ubuntu-18.04)"
-    runs-on: ubuntu-18.04
-    steps:
-      - uses: actions/checkout@v1
-      - run: sudo apt update
-      - name: "Install packages"
-        run: sudo apt-get install g++-8 libsdl1.2-dev libsdl-net1.2-dev libsdl-sound1.2-dev
-      - name: Build
-        env:
-          CC: gcc-8
-          CXX: g++-8
-        run: |
-          # build steps
-          set -x
-          $CXX --version
-          ./autogen.sh
-          ./configure CXXFLAGS="-Wall -fdiagnostics-color=always"
-          make -j "$(nproc)" |& tee build.log
-      - name: Summarize warnings
-        run: python3 ./scripts/count-warnings.py build.log
-
-  build_ubuntu:
+  build_gcc_ubuntu:
     name: "GCC"
     runs-on: ${{ matrix.os }}
```

Comment:
Is this matrix.os a type of meta/loop that goes through all the Ubuntu versions, and now included gcc-8, so that's why it was knocked out above?

Reply:
No - the reason is different. 

GCC8 was defined as a separate job (otherwise I would need to add a bunch of conditional statements in here and it turned out unreadable), but it was not providing additional value: we have this matrix'd job for the default Ubuntu compiler (GCC7 on 18.04, GCC5.4 on 16.04)  and GCC9 job for the newest available version of GCC… and this GCC8 was sitting between the two, without adding any benefit (it's non-default and finds fewer warnings than GCC9). I would rather have several new jobs targetting different architectures than redundant GCC8 for x86_64.

Matrix is GitHub's way of reusing the same job configuration across OSes - it works well for projects where it's possible to have 1 job definition. I tried to use it to have definition shared at least between default Ubuntus and MacOS, but it does not work too well in practice for DOSBox. Hopefully this matrix feature will be expanded in future - at the moment it only supports changing OS and nodejs version used for a build.

Reply:
Very interesting; thanks for the detailed explanation. I also agree on the GCC front - we've got plenty of coverage now on all of of the stable/odd versions. I'm really excited about the clang and Windows builds; this is already producing valuable fixes across several packages.